### PR TITLE
first test fct added for data processing

### PR DESCRIPTION
**Pull request by allaway**
This PR adds 4 new functions: 
2 functions (1 main + 1 helper) to take a samplesheet and check against the filenames to ensure that _1 and _2 were properly assigned in the sample sheet. This could be run before or after the workflow run. Honestly, not sure of the implications of swapping these - I've not done it before afaik - it might not matter, but on the off chance that it does matter for one or more of the many tools in sarek/rnaseq it is a good check to do. We could also repurpose this function to automatically annotate readPairs based on the name.

2 to help messaging to the console. These were fun... :)

Can test with: 
```
> samples <- check_readpair_validity('syn39542932') #should fail (artificial example)
> samples <- check_readpair_validity('syn29530880') #should pass
```

Known shortcomings of this function:
- I used portal data to figure out what the patterns/regexes should be. There's probably more possibilities, e.g. files that end in fq.gz or .fq instead of .fastq.gz or .fastq. 
- If the filename is wrong, it doesn't look at readPair for a second opinion. However, that's also a data integrity error IMO, so it's fine that these would get flagged as a mismatch.

### CODE REVIEW

File: R/nextflow_testing_utils.R [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+#'
+#' Read pairs are often encoded in the name of the file. Here, we check that if encoded in the name of the file,
+#' that the samplesheet read pair (e.g. _1 or _2) matches
+#'
+#' @param samplesheet A local file or syn id of samplesheet.
+#' @param parse_fun Function implementing how to parse samples in samplesheet.
+#' @import data.table
+#' @export
+#'
+#' @examples check_readpair_validity('syn39542932')
```
Comment (anngvu) [inline R/nextflow_testing_utils.R]: R-CMD check tries to run this example, causing error note. Wrap in /dontrun like here: https://github.com/nf-osi/nfportalutils/blob/develop/R/add_publication_from_pubmed.R#L63

File: R/nextflow_testing_utils.R [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+
+
+#' Format a test fail message.
+#'
+#' @param display_string A character string to format as a test failed message.
+#' @returns A message to the console
+#'
+test_failed <- function(display_string){
+  message(glue::glue("{emoji('broken_heart')} {bold(red('Test failed:'))} {red(display_string)}"))
```
Comment (anngvu) [inline R/nextflow_testing_utils.R]: Probably need to mention emoji pkg DESCRIPTION Imports (I get `error in emoji("broken_heart") : could not find function "emoji"`)

**Review by anngvu — CHANGES_REQUESTED** [review]

File: R/nextflow_testing_utils.R (hunk 1 above)
Comment (allaway) [reply R/nextflow_testing_utils.R]: Ooops, thanks! fixed.

File: R/nextflow_testing_utils.R (hunk 2 above)
Comment (allaway) [reply R/nextflow_testing_utils.R]: 😵‍💫 oops. fixed!

**Comment by anngvu** [comment]
@allaway Hey, so used this with a [real samplesheet here](https://www.synapse.org/#!Synapse:syn30841441) (because I realized I shouldn't just run the examples...) and got:
```
<U+0001F494> Test failed: Samplesheet read pair mismatch detected, check output.
# A tibble: 522 x 5
   sample                       input_id    readPair named_rp readpair_matches
   <chr>                        <chr>          <int>    <dbl> <lgl>           
 1 2-001_Plexiform_Neurofibroma syn15262441        1        1 TRUE            
 2 2-001_Plexiform_Neurofibroma syn15266271        2        1 FALSE           
 3 2-001_Plexiform_Neurofibroma syn15270018        3        1 FALSE           
 4 2-001_Plexiform_Neurofibroma syn15590067        4        1 FALSE           
 5 2-001_Plexiform_Neurofibroma syn15590155        5        1 FALSE           
 6 2-001_Plexiform_Neurofibroma syn15590278        6        1 FALSE           
 7 2-001_Plexiform_Neurofibroma syn15590341        7        1 FALSE           
 8 2-001_Plexiform_Neurofibroma syn15590404        8        1 FALSE           
 9 2-001_Plexiform_Neurofibroma syn15262492        9        2 FALSE           
10 2-001_Plexiform_Neurofibroma syn15266357       10        2 FALSE 
```
```
> table(x$readpair_matches)

FALSE  TRUE 
  465    57
```
That seems like a lot. Not sure how to interpret.

**Comment by allaway** [comment]
Thanks for the test! I'll take a look. I am guessing this has something to do with how I summarize the samplesheet. This one has multiple lanes/rows per sample, which I thought would be handled correctly but I guess not.

**Comment by allaway** [comment]
Alrighty @anngvu this should be fixed with the latest commit!

**Review by anngvu — APPROVED** [review]
👍🏼

**Pull request merged**